Mathlib/Data/Matrix/Rank.lean

```diff
@@ -114,15 +207,47 @@ theorem rank_submatrix_le [StrongRankCondition R] [Fintype m] (f : n → m) (e :
     LinearEquiv.range, Submodule.map_top]
   exact Submodule.finrank_map_le _ _
 
-theorem rank_reindex (e₁ : l ≃ m) (e₂ : n ≃ o) (A : Matrix l n R) :
-    rank (reindex e₁ e₂ A) = rank A := by
+@[simp]
+theorem rank_reindex [Fintype n₀] (em : m ≃ m₀) (en : n ≃ n₀) (A : Matrix m n R) :
+    rank (A.reindex em en) = rank A := by
   rw [rank, rank, mulVecLin_reindex, LinearMap.range_comp, LinearMap.range_comp,
     LinearEquiv.range, Submodule.map_top, LinearEquiv.finrank_map_eq]
 
 @[simp]
-theorem rank_submatrix (A : Matrix l n R) (e₁ : m ≃ l) (e₂ : o ≃ n) :
-    rank (A.submatrix e₁ e₂) = rank A :=
-  rank_reindex e₁.symm e₂.symm A
+theorem rank_submatrix [Fintype n₀] (A : Matrix m n R) (em : m₀ ≃ m) (en : n₀ ≃ n) :
+    rank (A.submatrix em en) = rank A := by
+  simpa only [reindex_apply] using rank_reindex em.symm en.symm A
+
+theorem lift_cRank_reindex {n : Type un} (A : Matrix m n R) (em : m ≃ m₀) (en : n ≃ n₀) :
+    lift.{um} (cRank (A.reindex em en)) = lift.{um₀} (cRank A) :=
+  (A.lift_cRank_submatrix_le em.symm en.symm).antisymm
+    <| by simpa using ((A.reindex em en).lift_cRank_submatrix_le em en)
+
+/-- A special case of `lift_cRank_reindex` for when the row types are in the same universe. -/
+@[simp]
+theorem cRank_reindex {m₀ : Type um} {n : Type un} (A : Matrix m n R) (em : m ≃ m₀) (en : n ≃ n₀) :
+    cRank (A.reindex em en) = cRank A := by
+  simpa using A.lift_cRank_reindex em en
+
+theorem lift_cRank_submatrix {n : Type un} (A : Matrix m n R) (em : m₀ ≃ m) (en : n₀ ≃ n) :
```

Comment:
```suggestion
@[simp]
theorem lift_cRank_submatrix {n : Type un} (A : Matrix m n R) (em : m₀ ≃ m) (en : n₀ ≃ n) :
```

Reply:
I made this change, but it breaks two of the other `@[simp]` proofs by oversimplifying to `True`, so it has to be explicitly excluded from those simp calls. I don't have a good sense for the right decision in such cases.

Reply:
I think the explicit exclusion is fine, especially since you're in the process of setting up the API.